src/test/common/configSettings.multiroot.test.ts

```diff
@@ -0,0 +1,181 @@
+        await settings.update('pythonPath', 'privatePythonPath', ConfigurationTarget.WorkspaceFolder);
+
+        const cfgSetting = PythonSettings.getInstance(workspaceUri);
+        assert.equal(cfgSetting.pythonPath, 'privatePythonPath', 'Python Path for workspace folder is incorrect');
+    });
+
+
+    test('Workspace folder should inherit Python Path from workspace root when opening a document', async () => {
+        const workspaceUri = Uri.file(path.join(multirootPath, 'workspace1'));
+        const fileToOpen = path.join(multirootPath, 'workspace1', 'file.py');
+
+        const settings = workspace.getConfiguration('python', workspaceUri);
+        const pythonPath = `x${new Date().getTime()}`;
+        await settings.update('pythonPath', pythonPath, ConfigurationTarget.Workspace);
+
+        const document = await workspace.openTextDocument(fileToOpen);
+        const cfg = PythonSettings.getInstance(document.uri);
+        assert.equal(cfg.pythonPath, pythonPath, 'Python Path not inherited from workspace');
+    });
+
+    test('Workspace folder should not inherit Python Path from workspace root when opening a document', async () => {
+        const workspaceUri = Uri.file(path.join(multirootPath, 'workspace1'));
+        const fileToOpen = path.join(multirootPath, 'workspace1', 'file.py');
+
+        const settings = workspace.getConfiguration('python', workspaceUri);
+        const pythonPath = `x${new Date().getTime()}`;
+        await settings.update('pythonPath', pythonPath, ConfigurationTarget.Workspace);
+        await settings.update('pythonPath', 'privatePythonPath', ConfigurationTarget.WorkspaceFolder);
+
+        const document = await workspace.openTextDocument(fileToOpen);
+        const cfg = PythonSettings.getInstance(document.uri);
+        assert.equal(cfg.pythonPath, 'privatePythonPath', 'Python Path for workspace folder is incorrect');
+    });
+
+    test('Enabling/Disabling Pylint in root should be reflected in config settings', async () => {
+        const workspaceUri = Uri.file(path.join(multirootPath, 'workspace1'));
+        await testLinterSetting(workspaceUri, ConfigurationTarget.Workspace, 'pylintEnabled', true);
+        await testLinterSetting(workspaceUri, ConfigurationTarget.Workspace, 'pylintEnabled', false);
+    });
+
+    test('Enabling/Disabling Pylint in root should be reflected in config settings', async () => {
+        const workspaceUri = Uri.file(path.join(multirootPath, 'workspace1'));
+        await testLinterSetting(workspaceUri, ConfigurationTarget.Workspace, 'pylintEnabled', true);
+        await testLinterSetting(workspaceUri, ConfigurationTarget.Workspace, 'pylintEnabled', false);
+    });
+
+    test('Enabling/Disabling Pylint in root and workspace should be reflected in config settings', async () => {
+        const workspaceUri = Uri.file(path.join(multirootPath, 'workspace1'));
+        const workspaceFolder = Uri.file(path.join(multirootPath, 'workspace1'));
```

Comment:
How is `workspaceFolder` different from `workspaceUri`? The code looks the same.

Reply:
wtf

Reply:
Fixed